django/db/models/sql/compiler.py

```diff
@@ -368,7 +368,12 @@ def get_order_by(self):
                     if is_ref and col_alias == src.refs:
                         src = src.source
                     elif col_alias:
-                        continue
+                        unresolved_src = expr.get_source_expressions()[0]
+                        # Check that the column alias is not equal to
+                        # the column name referenced in a F object (#31614)
+                        if not (hasattr(unresolved_src, "name") and unresolved_src.name == col_alias):
```

Comment:
It's the only way I found to access the column name referenced in the `F` object, but I don't fully understand all the expression objects, so maybe I could have used the `resolved` or `src` objects